tests/test_runner.py

```diff
@@ -598,3 +600,105 @@ def test_print_logs_flag_with_iterations():
     assert test_log_pos < test_error_pos
 
     assert ps.stderr == '', Tests.assertion_info('no errors', ps.stderr)
+
+## automatic database reconnection
+# TODO: This integration test should be moved to a dedicated integration test suite once that structure is implemented (https://github.com/green-coding-solutions/green-metrics-tool/issues/1302) # pylint: disable=fixme
+def test_database_reconnection_during_run_integration():
+    """Integration test: Verify GMT runner handles database reconnection during execution
+    
+    This test simulates a database outage by restarting the postgres container mid-run.
+    With database retry logic implemented, the test should succeed by automatically
+    reconnecting when the database becomes available again.
+    
+    Timing analysis based on a debug run with logs:
+    T+0.0s  - GMT runner starts, database restart thread starts waiting
+    T+5.0s  - RUNTIME phase begins (with 15 seconds sleep command)
+    T+12.3s - Database restart occurs (during runtime phase)
+    T+13.1s - Database restart completes
+    T+16.1s - Database should be available again
+    T+20.0s - REMOVE phase starts (database operations resume with retry logic)
+    T+22.4s - Test completes successfully with database reconnection
+    
+    This timing ensures database restart happens during active measurement phase
+    when database operations are likely occurring for metric storage.
+    """
+
+    def log_with_timestamp(message, prefix="TEST", start_time=None):
+        """Helper function to log messages with timestamp and optional elapsed time"""
+        current_time = time.time()
+        timestamp = time.strftime('%H:%M:%S', time.localtime(current_time))
+        if start_time:
+            elapsed = f" (T+{current_time-start_time:.1f}s)"
+        else:
+            elapsed = ""
+        print(f"[{timestamp}] [{prefix}] {message}{elapsed}")
+
+    run_name = 'test_db_reconnect_' + utils.randomword(12)
+    test_start_time = time.time()
+
+    def restart_database():
+        # Restart database during metrics collection/storage phase
+        log_with_timestamp("Waiting 15 seconds before restarting database...", "DB RESTART")
+        time.sleep(15)  # Wait for runtime to start but not complete
+        log_with_timestamp("Restarting test-green-coding-postgres-container now...", "DB RESTART", test_start_time)
+        result = subprocess.run(['docker', 'restart', 'test-green-coding-postgres-container'],
+                               check=True, capture_output=True)
+        log_with_timestamp(f"Database restart completed. Docker output: {result.stdout.decode().strip()}", "DB RESTART", test_start_time)
+        time.sleep(3)  # Give DB time to restart
+        log_with_timestamp("Database should be available again after 3s wait", "DB RESTART", test_start_time)
+
+    # Start database restart in background thread
+    restart_thread = threading.Thread(target=restart_database)
```

Comment:
I wonder if this is the nicest implementation. If the sole reason is that docker restart takes long to return I think a docker compose stop -t 0 and then a docker a docker start should do it. Since there should be no I/O also no thread is needed.

Would that help?

My issue with the thread: Everytime I see the Thread library used my brain switches to scanner mode to find the high complexity part of the code. So it makes code for me harder to read if there is no strong reason for it to use. Maybe also a me issue ...